### ISSUE TITLE
OpenHue-CLI Discovery Never Resolves on Windows

### ISSUE THREAD

**Issue by Xevion** [openhue/openhue-cli#52]
At least, I assume it never will, because on Linux, this process finished in half a second, while on **Windows**, it's still running.

I'm building with `go install` directly instead of docker.

Everything else about the CLI works fine as far as I can tell, I just had to setup by manually checking and feeding in the Bridge IP address; `openhue-cli setup -b 192.168.100.152`.

I enabled verbose logging by setting the dev environment variable, but the logs produced don't make it obvious to me what the issue is:

```
PS C:\Users\Xevion\projects\openhue-cli> openhue-cli discover
time="2024-05-19T00:53:33-05:00" level=info msg="Running in dev mode, logs will be output in the console"
time="2024-05-19T00:53:33-05:00" level=info msg="Running the 'discover' command"
time="2024-05-19T00:53:33-05:00" level=info msg="DNS-SD service '_hue._tcp.local.'"
time="2024-05-19T00:53:38-05:00" level=error msg="Unable to lookup bridge with service '_hue._tcp.local.'. Check error details below:"
time="2024-05-19T00:53:38-05:00" level=error msg="context deadline exceeded"
```

I waited a few minutes to see if anything else was outputted, but nothing did.

- openhue-cli [master](https://github.com/openhue/openhue-cli/tree/c42df59b426f4bd91805c14b37cc96fb158a5ee8) (c42df59)
- go1.22.0 windows/amd64
- Windows 10

**Comment by thibauult** [comment]
Hi @Xevion and thanks for reporting this issue. 

My main problem in investigating this one is that I don´t have any Windows machine to test with. 

Would you be able to run the following command on your side: 
```
dns-sd -B _hue._tcp local.
```
> [!NOTE]
> Command found from the [Philips Hue Developers documentation](https://developers.meethue.com/develop/application-design-guidance/hue-bridge-discovery/)

**Comment by Xevion** [comment]
```
PS C:\Users\Xevion> dns-sd -B _hue._tcp local.
Browsing for _hue._tcp.local.
Timestamp     A/R Flags if Domain                    Service Type              Instance Name
 2:46:23.291  Add     2 20 local.                    _hue._tcp.                Hue Bridge - 98C048
```

I had to install the Bonjour SDK for Windows from Apple, which was much harder to find than it should be.

FYI for anyone exploring this in the future, you need an Apple ID (standard, no special developer setup/enrollment required).
- https://developer.apple.com/bonjour/ (**Bonjour for Windows**, click **Bonjour SDK for Windows**, sign-in)
- Final link: https://developer.apple.com/download/all/?q=Bonjour%20SDK%20for%20Windows
- Install `bonjoursdksetup.exe`, `dns-sd` will then be available in PATH.

And actually, after looking into Bounjour, I found that [Bonjour Print Services](https://support.apple.com/en-us/106380) actually does install `dns-sd` into path (confirmed it).

I looked into `Resolve-DnsName` as a more readily available alternative, but it doesn't seem to be able to look up service instances (qualified by the `-B` argument). I also found a npm package, [node-dns-sd](https://www.npmjs.com/package/node-dns-sd), and it actually does work pretty well. It has a CLI wrapper, details here: [@warren-bank/node-dns-sd-cli](https://github.com/warren-bank/node-dns-sd-cli/blob/master/README.md).

**Comment by thibauult** [comment]
Hi @Xevion, 

I've just released the [0.16](https://github.com/openhue/openhue-cli/releases/tag/0.16), including a few improvements regarding the automatic bridge discovery.  

Please let me know if the `openhue discover` command is now working better in Windows. 

Thanks!

**Comment by Xevion** [comment]
Apologies for the latency, but I tested and confirmed that discovery is working on Windows.

```
PS C:\Users\Xevion\Downloads\openhue-cli> go run . setup
[OK] Found Hue Bridge with IP '192.168.100.152'
[..] Please push the button on your Hue Bridge
.........
[OK] Successfully paired openhue with your Hue Bridge!
[OK] Configuration saved in file C:\Users\Xevion\.openhue\config.yaml
```
